man/openmp-utils.Rd

```diff
@@ -3,24 +3,26 @@
 \alias{getDTthreads}
 \title{ Set or get number of threads that data.table should use }
 \description{
-Set and get number of threads to be used in \code{data.table} functions that are parallelized with OpenMP. Default value 0 means to utilize all CPU available with an appropriate number of threads calculated by OpenMP. \code{getDTthreads()} returns the number of threads that will be used. This affects \code{data.table} only and does not change R itself or other packages using OpenMP.  We have followed the advice of section 1.2.1.1 in the R-exts manual: "\ldots or, better, for the regions in your code as part of their specification\ldots num_threads(nthreads)\ldots That way you only control your own code and not that of other OpenMP users."  All the parallel region in data.table contain this directive. This is mandated by a \code{grep} in data.table's quality control CRAN release procedure script.
+  Set and get number of threads to be used in \code{data.table} functions that are parallelized with OpenMP.
 }
 \usage{
-setDTthreads(threads = 0, restore_after_fork = NULL)
-getDTthreads(verbose = getOption("datatable.verbose", FALSE))
+  setDTthreads(threads = 0, restore_after_fork = NULL)
+  getDTthreads(verbose = getOption("datatable.verbose", FALSE))
 }
 \arguments{
-  \item{threads}{ An integer >= 0. Default 0 means use all CPU available and leave the operating system to multi task. }
+  \item{threads}{ An integer >= 0. Default and recommended 0 means to use half of the logical CPUs available so as to leave plenty of resources for other processes. }
```

Comment:
Would be useful to provide user option to use all cores without forcing them to figure out how many logical cores they have. Maybe `threads=Inf`?

Reply:
Perhaps it would be better to have `threads = NULL` be the default, keep `threads = 0` to retain the previous behaviour, and special-case `if (is.null(threads)) threads = <half of CPU> else threads`. That way, people relying on `threads = 0` to work as documented can continue to, but people just using 'the data.table default' are moved to the new default.